clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -7430,6 +7430,212 @@ static void handleSYCLIntelFPGAMaxConcurrencyAttr(Sema &S, Decl *D,
   S.AddSYCLIntelFPGAMaxConcurrencyAttr(D, A, E);
 }
 
+// Checks if an expression is a valid filter list for an add_ir_attributes_*
+// attribute. Returns true if an error occured.
+static bool checkAddIRAttributesFilterListExpr(Expr *FilterListArg, Sema *S,
```

Comment:
In all of these static helper functions `Sema *S` should be `Sema &S` (there's never a reason for the `Sema` object to be null).